syntax = "proto3";

package serialize;

import "map_renderer.proto";
import "transport_router.proto";

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    string name = 1;
    Coordinates coord = 2;
    uint32 id = 3;
}

message Bus {
    string name = 1;
    bool is_ring = 2;
    repeated uint32 route = 3;
    uint32 id = 4;
}

message Distance {
    uint32 from_id = 1;
    uint32 to_id = 2;
    int32 distance = 3;
}

message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated Distance distances = 3;
    RenderSettings render_settings = 4;
    TransportRouter router = 5;
}